Content.Server/Interaction/InteractionSystem.cs

```diff
@@ -202,6 +204,8 @@ public override void DoAttack(EntityUid user, EntityCoordinates coordinates, boo
             {
                 var item = hands.ActiveHandEntity;
 
+                if (HasComp<ServerRangedBarrelComponent>(item)) return;
```

Comment:
this is a little balls, is there a plan for something better

Reply:
Talked to electro and probably just making an event.